frame/support/procedural/src/pallet/expand/type_value.rs

```diff
@@ -16,38 +16,55 @@
 // limitations under the License.
 
 use crate::pallet::Def;
-use syn::spanned::Spanned;
 
 /// * Generate the struct
 /// * implement the `Get<..>` on it
+/// * Rename the name of the function to internal name
 pub fn expand_type_values(def: &mut Def) -> proc_macro2::TokenStream {
 	let mut expand = quote::quote!();
 	let frame_support = &def.frame_support;
 
 	for type_value in &def.type_values {
-		// Remove item from module content
-		let item = &mut def.item.content.as_mut().expect("Checked by def").1[type_value.index];
-		let span = item.span();
-		*item = syn::Item::Verbatim(Default::default());
+		let fn_ident_renamed = syn::Ident::new(
+			&format!("__type_value_for_{}", type_value.ident),
+			type_value.ident.span(),
+		);
+
+		let type_value_item = {
+			let item = &mut def.item.content.as_mut().expect("Checked by def").1[type_value.index];
+			if let syn::Item::Fn(item) = item {
+				item
+			} else {
+				unreachable!("Checked by error parser")
+			}
+		};
+
+		// Rename the type_value function name
+		type_value_item.sig.ident = fn_ident_renamed.clone();
+		type_value_item.attrs.push(syn::parse_quote!(#[allow(non_snake_case)]));
```

Comment:
Why not https://docs.rs/Inflector/0.11.4/inflector/cases/snakecase/fn.to_snake_case.html ?